### ISSUE TITLE
xtext-xtend-api-diff broken

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#1519]
https://ci-staging.eclipse.org/xtext/job/releng/job/xtext-xtend-api-diff/25/console
```
Started by timer
Running as SYSTEM
[EnvInject] - Loading node environment variables.
Building remotely on external-agent in workspace /home/xtextbot/workspace/releng/xtext-xtend-api-diff
No credentials specified
 > /usr/bin/git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > /usr/bin/git config remote.origin.url https://github.com/xtext/xtext-apidiff.git # timeout=10
Fetching upstream changes from https://github.com/xtext/xtext-apidiff.git
 > /usr/bin/git --version # timeout=10
 > /usr/bin/git fetch --tags --progress https://github.com/xtext/xtext-apidiff.git +refs/heads/*:refs/remotes/origin/*
 > /usr/bin/git rev-parse refs/remotes/origin/master^{commit} # timeout=10
 > /usr/bin/git rev-parse refs/remotes/origin/origin/master^{commit} # timeout=10
Checking out Revision 87d7e15ff1aa04f273a7a4ec82143e1f227a4029 (refs/remotes/origin/master)
 > /usr/bin/git config core.sparsecheckout # timeout=10
 > /usr/bin/git checkout -f 87d7e15ff1aa04f273a7a4ec82143e1f227a4029
Commit message: "Merge pull request #3 from xtext/cd_fixTheJob"
 > /usr/bin/git rev-list --no-walk 87d7e15ff1aa04f273a7a4ec82143e1f227a4029 # timeout=10
Script returned: #25 (2.18.0->2.19.0)
[xtext-xtend-api-diff] $ /bin/sh -xe /tmp/jenkins4836343534568423861.sh
+ ./create-api-diff.sh

ERROR: JAVA_HOME is set to an invalid directory: /opt/tools/java/oracle/jdk-8/latest

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.

Diffing 2.19.0 against 2.18.0
cp: cannot stat 'xtext-apidiff/build/libs/japicmp-ext.jar': No such file or directory
Downloading Xtext 2.17.0
./create-api-diff.sh: line 45: unzip: command not found
Downloading Xtext 2.17.1
./create-api-diff.sh: line 53: unzip: command not found
Downloading Xtext 2.18.0
./create-api-diff.sh: line 61: unzip: command not found
Downloading Xtext 2.19.0
./create-api-diff.sh: line 72: unzip: command not found
mv: cannot stat 'p2.repository': No such file or directory
creating the actual diff
Error: Unable to access jarfile japicmp-ext.jar
Build step 'Execute shell' marked build as failure
Archiving artifacts
[htmlpublisher] Archiving HTML reports...
[htmlpublisher] Archiving at PROJECT level /home/xtextbot/workspace/releng/xtext-xtend-api-diff/output to /var/jenkins_home/jobs/releng/jobs/xtext-xtend-api-diff/htmlreports/Diff_20Report
ERROR: Specified HTML directory '/home/xtextbot/workspace/releng/xtext-xtend-api-diff/output' does not exist.
Finished: FAILURE

```

**Comment by cdietrich** [comment]
@Hejado

> Building remotely on external-agent in workspace /home/xtextbot/workspace/releng/xtext-xtend-api-diff 

seems to run on the external server but there seems missing configuration

**Comment by Hejado** [comment]
=> do we need java and co on our server?

Yes, and java is installed and running (because there is an agent process). I believe, `JAVA_HOME` is not set by default when installing java. So setting this ENV correctly should fix that, but...

> shall the call to the script be moved to a container?

... we should use containers so we can utilize our agents as similar as possible.

Possible relevant: https://bugs.eclipse.org/bugs/show_bug.cgi?id=549184

**Comment by cdietrich** [comment]
Fixed in 2.19
